### PR TITLE
Auto forwarder implementation

### PR DESCRIPTION
**Pull request by esinarta**
- Re-activated AutoForwarder and Docker UI components 
- Added default value for endpoint picker rather than having one selected automatically
- Implemented auto-forwarding inside AutoForwarder component
- Created new AutoForwarderWebhookCreated type for GraphQL subscription

### CODE REVIEW

File: src/routes/Viewport/AutoForwarder/index.tsx [hunk 1]
```diff
@@ -22,6 +36,18 @@ const AutoForwarder = ({ docked }: { docked: Boolean }) => {
   const [running, setRunning] = useState(false);
   const { forwardUrls } = useForwardUrls(endpointId);
   const [url, setUrl] = useState('');
+  const { forwardWebhook } = useForwarder(endpointId);
```
Comment (johnnyoshika) [inline src/routes/Viewport/AutoForwarder/index.tsx:23]: When endpointId is empty (i.e. ''), `useForwardUrls(endpointId)` is causing a GraphQL error, as it tries to fetch forward URLs for an endpoint that's empty.  Here is the request that's being sent:

image

This results in this response from our GraphQL server:

image

Also, every time I change the endpoint from the dropdown, the same request is being sent with an empty endpointId. You can test this by putting a breakpoint here, as that code will be called on every GraphQL error: https://github.com/webhook-buddy/webhook-buddy-client/blob/master/src/index.tsx#L72

You can also see this error being printed to the console:
image

To fix this, I suggest you extract this into a separate component, and only load it when an appropriate endpoint is selected:
```
        <label>Auto-forward to</label>
        <Autosuggest
          type="url"
          placeholder="Forward to URL (e.g. http://localhost:8000/send-webhook-here)"
          userInput={url}
          setUserInput={setUrl}
          suggestions={forwardUrls}
          position={AutosuggestPositionEnum.Up}
          disabled={running}
        />
      </div>
```

You may also want to extract the endpoint select list to a separate component as well to make it consistent.

**Review by johnnyoshika — CHANGES_REQUESTED** [review]

**Comment by johnnyoshika** [comment]
@esinarta Before you make further changes, can you rebase on top of this main repo's master branch? It looks like your branch hasn't picked up the latest changes. If you haven't set up this repo as a remote repo, you can do something like this:

```
git remote add source https://github.com/webhook-buddy/webhook-buddy-client.git
git fetch source
git rebase source/master
git push -f
```

**Comment by esinarta** [comment]
> @esinarta Before you make further changes, can you rebase on top of this main repo's master branch? It looks like your branch hasn't picked up the latest changes. If you haven't set up this repo as a remote repo, you can do something like this:
> 
> ```
> git remote add source https://github.com/webhook-buddy/webhook-buddy-client.git
> git fetch source
> git rebase source/master
> git push -f
> ```

@johnnyoshika Okay, I just followed these steps to rebase.

**Comment by BuzzBumble** [comment]
@johnnyoshika  

> Also, every time I change the endpoint from the dropdown, the same request is being sent with an empty endpointId. 

We couldn't seem to recreate this bug (maybe we fixed it inadvertently?)

> When endpointId is empty (i.e. ''), useForwardUrls(endpointId) is causing a GraphQL error, as it tries to fetch forward URLs for an endpoint that's empty.

This is now fixed, since the AutoSuggest is now in its own conditionally-rendered component and the useForwardURLs hook is being used within that component.

The last concern is that when a webhook is auto-forwarded, the profiler is noticing 10+ re-renders of the AutoForwarder's child components (namely `AutoForwardSuggest`). Still looking into this.

**Comment by johnnyoshika** [comment]
> The last concern is that when a webhook is auto-forwarded, the profiler is noticing 10+ re-renders of the AutoForwarder's child components (namely `AutoForwardSuggest`). Still looking into this.

Thanks @BuzzBumble. Would you like me to wait until the rendering issues are resolved or proceed with the review? We can also review the render problem together tomorrow.

**Comment by johnnyoshika** [comment]
A couple of issues:

* I'm still seeing the `FORBIDDEN error` when I switch endpoints due to the empty `endpointId` being passed to `forwardUrls` query:

image

* I think you can delete everything inside `src/hooks/types`. Our types generation tool doesn't delete existing types. Try deleting that folder and then re-run the type generation. If the deleted ones don't get re-recreated, they were safe to delete.

File: src/routes/Viewport/AutoForwarder/index.tsx [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 import { GetEndpoints } from 'schema/types/GetEndpoints';
-import Error from 'components/Error';
-import useForwardUrls from 'hooks/useForwardUrls';
-import Autosuggest, {
-  AutosuggestPositionEnum,
-} from 'components/Autosuggest';
-
+import useForwarder from 'hooks/useForwarder';
 import styles from './styles.module.css';
+import { WebhookCreated } from 'hooks/types/WebhookCreated';
```
Comment (johnnyoshika) [inline src/routes/Viewport/AutoForwarder/index.tsx]: This is referencing types inside `hooks/types` instead of `schema/types`

**Review by johnnyoshika — CHANGES_REQUESTED** [review]

**Comment by esinarta** [comment]
I deleted the hooks/types folder and re-ran types-generate and none of the types were re-created. I also changed the WebhookCreated import to reference schema/types.

**Review by johnnyoshika — APPROVED** [review]

**Comment by johnnyoshika** [comment]
Thanks for the PR. Well done, guys!

**Pull request merged**